api/v1alpha1/harborconfiguration_types.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&HarborConfiguration{}, &HarborConfigurationList{})
+}
+
+type HarborConfigurationSpec struct {
+	HarborTarget HarborTarget `json:"harborTarget,omitempty"`
+	Registry     Registry     `json:"registry,omitempty"`
+	ProjectReq   ProjectReq   `json:"projectReq,omitempty"`
+	Replication  Replication  `json:"replication,omitempty"`
+}
+
+type HarborConfigurationStatus struct {
+	RegistryId    int64  `json:"registryId,omitempty"`
+	ProjectId     string `json:"projectId,omitempty"`
+	ReplicationId int64  `json:"replicationId,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+type HarborConfiguration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   HarborConfigurationSpec   `json:"spec,omitempty"`
+	Status HarborConfigurationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+type HarborConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []HarborConfiguration `json:"items,omitempty"`
+}
+
+type HarborTarget struct {
```

Comment:
I would still like this to point to the harbor CR instead of the API url - so we can read all necessary info from the harbor operator CR.

Reply:
Do you mean the HarborTarget struct? or the Items?

Reply:
No the HarborTarget struct - so it could be something like 
```
name
namespace
```
Of a harbor CR